### ISSUE TITLE
DOC: fix RT03 errors in docstrings 

### ISSUE THREAD

**Issue by jordan-d-murphy** [pandas-dev/pandas#57416]
Pandas has a script for validating docstrings:

https://github.com/pandas-dev/pandas/blob/44c50b20e8d08613144b3353d9cd0844a53bd077/ci/code_checks.sh#L144-L414

Currently, some methods fail the RT03 check.

The task here is:

1. take 2-4 methods
2. run: scripts/validate_docstrings.py --format=actions --errors=RT03 `method-name`
3. check if validation docstrings passes for those methods, and if it’s necessary fix the docstrings according to whatever error is reported
4. remove those methods from code_checks.sh
5. commit, push, open pull request

Please don't comment take as multiple people can work on this issue. You also don't need to ask for permission to work on this, just comment on which methods are you going to work.

If you're new contributor, please check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)

**Comment by jordan-d-murphy** [comment]
I don't have permission to add them, but this could probably use some labels:

- CI
- Docs
- good first issue

**Comment by jordan-d-murphy** [comment]
Addresses https://github.com/pandas-dev/pandas/pull/57356

**Comment by bergnerjonas** [comment]
Will work on 

- pandas.Index.to_numpy
- pandas.Categorical.set_categories

**Comment by bergnerjonas** [comment]
Continue with:

- pandas.CategoricalIndex.set_categories
- pandas.DataFrame.astype
- pandas.DataFrame.at_time
- pandas.DataFrame.ewm

**Comment by jordan-d-murphy** [comment]
opened a fix for `pandas.DataFrame.expanding`

**Comment by YashpalAhlawat** [comment]
opened a fix for pandas.read_sql

**Comment by YashpalAhlawat** [comment]
Opened a fix for `pandas.read_sql_query`, `pandas.read_feather`

**Comment by YashpalAhlawat** [comment]
Opened a fix for below methods:
`pandas.DataFrame.min`
`pandas.DataFrame.max`
`pandas.DataFrame.mean`
`pandas.DataFrame.median`
`pandas.DataFrame.skew`
`pandas.DataFrame.kurt`

**Comment by bergnerjonas** [comment]
Will work on:
pandas.DataFrame.expanding
pandas.DataFrame.filter
pandas.DataFrame.first_valid_index
pandas.DataFrame.get

**Comment by bergnerjonas** [comment]
Continue with:
pandas.DataFrame.nsmallest
 pandas.DataFrame.nunique
 pandas.DataFrame.pipe
 pandas.DataFrame.plot.box 
 pandas.DataFrame.plot.density
 pandas.DataFrame.plot.kde
 pandas.DataFrame.plot.scatter

**Comment by bergnerjonas** [comment]
Will work on:

- pandas.DataFrame.pop
- pandas.DataFrame.prod
- pandas.DataFrame.product
- pandas.DataFrame.reindex
- pandas.DataFrame.reorder_levels
- pandas.DataFrame.sem

**Comment by bergnerjonas** [comment]
https://github.com/pandas-dev/pandas/blob/dc19148bf7197a928a129b1d1679b1445a7ea7c7/ci/code_checks.sh#L615C1-L864C42

Current status: still 200+ method docstrings need to be fixed, maybe add the "good first issue" to get more people involved ;)

**Comment by jordan-d-murphy** [comment]
agreed! I don't have permissions to add labels, otherwise I would. hopefully a core team member can add that label for us. 

thanks for all your work on these!

**Comment by jordan-d-murphy** [comment]
opened a fix for 
```
pandas.timedelta_range 
pandas.util.hash_pandas_object
```

**Comment by jordan-d-murphy** [comment]
opened a. fix for 
```
pandas.read_orc
pandas.read_sas
pandas.read_spss
pandas.read_stata
```

**Comment by jordan-d-murphy** [comment]
Hi all, if https://github.com/pandas-dev/pandas/pull/57826 gets merged in, I might be reworking our approach here; this would look like closing the following issues: 

https://github.com/pandas-dev/pandas/issues/57443
https://github.com/pandas-dev/pandas/issues/57438
https://github.com/pandas-dev/pandas/issues/57420
https://github.com/pandas-dev/pandas/issues/57417
https://github.com/pandas-dev/pandas/issues/57416
https://github.com/pandas-dev/pandas/issues/57111

And opening a new issue to address these based on the new approach. 

tl;dr 

the work can still be done, but probably under a new ticket once  #57826 is merged in

**Comment by jordan-d-murphy** [comment]
Closed by:
[CI: speedup docstring check consecutive runs #57826](https://github.com/pandas-dev/pandas/pull/57826)
[CI: Better error control in the validation of docstrings #57879](https://github.com/pandas-dev/pandas/pull/57879) 

Opening a new issue to fix docstring errors.

**Comment by jordan-d-murphy** [comment]
Opened [DOC: Enforce Numpy Docstring Validation (Parent Issue) #58063](https://github.com/pandas-dev/pandas/issues/58063) as a parent issue for fixing docstrings based on the refactoring in code_checks.sh 

Feel free to swing by and help out! 🙂
